docs/topics/db/models.txt

```diff
@@ -844,6 +844,26 @@ built-in model methods, adding new arguments. If you use ``*args,
 **kwargs`` in your method definitions, you are guaranteed that your
 code will automatically support those arguments when they are added.
 
+If you wish to update a field in the :meth:`~Model.save` method, you may also
+want to have this field automatically added to the ``update_fields`` kwarg.
+See :doc:`specifying which fields to save </ref/models/instances>`.
+This will ensure the field is saved when ``update_fields`` is specified.
+For example::
+
+
+    from django.db import models
+    from django.utils.text import slugify
+
+    class Blog(models.Model):
+        name = models.CharField(max_length=100)
+        slug = models.TextField()
+
+        def save(self, *args, update_fields, **kwargs):
```

Comment:
This should have the default argument for `update_fields` otherwise it'll be forced to be required which might break things if anywhere calling `.save()` assumes it to be optional.
```suggestion
        def save(self, *args, update_fields=None, **kwargs):
```
However, there is also the issue that this current signature will force it to be keyword-only. Unfortunately, while it is likely that this will normally be passed as a keyword, it isn't forbidden for it to be positional. This makes things tricky. Potentially it is safest to just define the whole signature up to the required field that we need to handle, i.e. `update_fields`, and allow for any future additions by providing `*args, **kwargs` after:
```suggestion
        def save(
            self, force_insert=False, force_update=False, using=None, update_fields=None,
            *args, **kwargs,
        ):
```
Other than an incomplete example in [`docs/ref/contrib/sitemaps.txt`](https://github.com/django/django/blob/37c5b8c07be104fd5288cd87f101e48cb7a40298/docs/ref/contrib/sitemaps.txt#L608), there isn't really much precedent for the right approach here.

Maybe @carltongibson will have some opinion on this?